ssv/runner.go

```diff
@@ -155,30 +139,24 @@ func (b *BaseRunner) basePostConsensusMsgProcessing(runner Runner, signedMsg *ty
 	return hasQuorum, roots, errors.Wrap(err, "could not process post-consensus partial signature msg")
 }
 
-// basePartialSigMsgProcessing adds an already validated partial msg to the container, checks for quorum and returns true (and roots) if quorum exists
+// basePartialSigMsgProcessing adds an already validated partial msg to the container,
+// checks for quorum and returns true (and roots) if we should process the msg
 func (b *BaseRunner) basePartialSigMsgProcessing(
 	signedMsg *types.SignedPartialSignatureMessage,
-	container *PartialSigContainer,
+	container PartialSignatureContainer,
 ) (bool, [][32]byte, error) {
-	roots := make([][32]byte, 0)
-	anyQuorum := false
-	for _, msg := range signedMsg.Message.Messages {
-		prevQuorum := container.HasQuorum(msg.SigningRoot)
-
-		container.AddSignature(msg)
 
-		if prevQuorum {
-			continue
-		}
-
-		quorum := container.HasQuorum(msg.SigningRoot)
-		if quorum {
-			roots = append(roots, msg.SigningRoot)
-			anyQuorum = true
-		}
+	if b.Share.HasQuorum(len(container)) {
+		container[signedMsg.Signer] = signedMsg
+		// return false if we already have quorum
+		return false, container.Roots(), nil
 	}
 
-	return anyQuorum, roots, nil
+	container[signedMsg.Signer] = signedMsg
+	if b.Share.HasQuorum(len(container)) {
+		return true, container.Roots(), nil
+	}
```

Comment:
Isn't this skipping the check that the signers have signed the same Beacon roots?

I imagine that if operators sign different aggregation messages, and the signing roots aren't checked, then the Runner may falsely think it has quorum and proceed to fail reconstructing the signature. If true, this means that one faulty operator may ruin the duty, no?

On another note, it seems to me that `container.Roots()` may cause another consensus issue. Since it only returns the signing roots from one `SignedPartialSignatureMessage`, this means that if the operators have signed different roots, then we'd only try to reconstruct & submit those from the first message in the container (which is a map so the order is non-deterministic).

In case anyone wants to know why multiple signing roots are necessary:
in sync committee aggregation duties, although extremely rare, one validator may participate in several subcommittees, in which case there will be several roots under `signedMsg.Message.Messages`, one for each subcommittee.

Reply:
So double check me please @moshe-blox:

1. Note that before `basePartialSigMsgProcessing` is called we always call some validation method. See:
https://github.com/bloxapp/ssv-spec/blob/daec5450ce7d808a7e643081a5109b4eb7c55f07/ssv/pre_consensus_justification.go#L20
https://github.com/bloxapp/ssv-spec/blob/0a414236cef5eca4dc0f7da464034325869a6477/ssv/runner_validations.go#L29
https://github.com/bloxapp/ssv-spec/blob/0a414236cef5eca4dc0f7da464034325869a6477/ssv/runner_validations.go#L12

Those validations should take care of the problem you mentioned. I will add method comments though to make this clear though.

2. Having said this it might be smart when it comes to the implementation to add this check as part of a defensive programming strategy. If a bug is introduced in validation you have a fall back mechanism. Or you can add comments and have good tests (your choice, just giving my 2 cents.

3. Same with the problem concerning `container.Roots()`. Not sure if I need to add comments though because this is just a getter method to what was inserted to the container.

Reply:
actually @moshe-blox I think the existing comment is fine:
https://github.com/bloxapp/ssv-spec/blob/ecc6a3f944d3dcb2387be71eb29920276f11a131/ssv/runner.go#L142

If you think it should be clearer you can make a suggestion :-)

Reply:
hey @GalRogozinski 

Thanks for the clarifying, I see that this issue can't happen because the roots must be identical to what the verifier got from their own Beacon node.